File: leet-code/remove-adjacent.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        str = []
        i = 0
        
        while i < len(s):
            if len(str) !=0 and str[-1]==s[i]:
                i+=1
                str.pop(-1)
            else:
                str.append(s[i])
                i+=1
            
            return str

File: leet-code/remove-adjacent.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string answer;

        for(const char c: s) {
            if(answer.empty() || answer.back() != c) {
                answer.push_back(c);
            } else {
                answer.pop_back();
            }
        }

        return answer;
    }
};